packages/teams-js/src/private/files.ts

```diff
@@ -710,40 +731,90 @@ export namespace files {
    * @hidden
    * Hide from docs
    *
-   * Initiates the 3P cloud storage provider file action (Upload / Download / Delete) flow
+   * Initiates the delete 3P cloud storage file(s) flow, which will delete existing file(s) from the given 3P provider
+   *
+   * @param deleteFileRequest 3P cloud storage provider delete action request content
+   * @param callback Callback that will be triggered post deleting existing file(s) flow is finished
+   */
+  export function deleteCloudStorageProviderFile(
+    deleteFileRequest: CloudStorageProviderRequest<CloudStorageProviderDeleteFileContent>,
+    callback: (error?: SdkError) => void,
+  ): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!callback) {
+      throw getSdkError(ErrorCode.INVALID_ARGUMENTS, '[files.deleteCloudStorageProviderFile] callback cannot be null');
+    }
+
+    if (!(deleteFileRequest && deleteFileRequest.content)) {
+      throw getSdkError(
+        ErrorCode.INVALID_ARGUMENTS,
+        '[files.deleteCloudStorageProviderFile] 3P cloud storage provider request content is missing',
+      );
+    }
+
+    sendMessageToParent('files.deleteCloudStorageProviderFile', [deleteFileRequest], callback);
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
    *
-   * Upload Action : Allows uploading file(s) to the given 3P cloud storage provider
-   * Download Action : Allows downloading file(s) from the given 3P cloud storage provider
-   * Delete Action : Allows deleting file(s) from the given 3P cloud storage provider
+   * Initiates the download 3P cloud storage file(s) flow, which will download existing file(s) from the given 3P provider
    *
-   * @param cloudStorageProviderFileActionRequest 3P cloud storage provider file action (Upload / Download / Delete) request content
-   * @param callback Callback that will be triggered post 3P cloud storage action
+   * @param downloadFileRequest 3P cloud storage provider download file(s) action request content
+   * @param callback Callback that will be triggered post downloading existing file(s) flow is finished
    */
-  export function performCloudStorageProviderFileAction(
-    cloudStorageProviderFileActionRequest: CloudStorageProviderRequest<CloudStorageProviderActionContent>,
+  export function downloadCloudStorageProviderFile(
```

Comment:
How does download work exactly? I don't see the callback returning any file information

Reply:
Also, same comment from above about checking itemList

Reply:
Made validation on itemList. 

Download Working :
The consuming app (in our case from OD side) initiates the download of file(s) with required file metadata. SDK is just delegating the call and teams side takes care of download of the file(s). The reason callback has no information related to file is that the entire download process is taken care at teams side. Main reason behind this being 3P file access requiring authentication, infra for which is present at teams side only and not at OD side.